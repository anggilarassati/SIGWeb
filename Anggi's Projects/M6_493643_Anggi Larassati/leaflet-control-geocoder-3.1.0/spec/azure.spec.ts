import { afterEach, describe, expect, it, vi } from 'vitest';
import { GeocodingResult } from '../src/geocoders/api';
import { AzureMaps } from '../src/geocoders/azure';
import { mockFetchRequest } from './mockFetchRequest';

describe('L.Control.Geocoder.AzureMaps', () => {
  afterEach(() => vi.clearAllMocks());
  it('geocodes', async () => {
    const geocoder = new AzureMaps({ apiKey: 'xxx' });
    const result = await mockFetchRequest(
      'https://atlas.microsoft.com/search/address/json?api-version=1.0&query=15127+NE+24th+Street%2C+Redmond%2C+WA+98052&subscription-key=xxx',
      {
        summary: {
          query: '15127 NE 24th Street, Redmond, WA 98052',
          queryType: 'NON_NEAR',
          queryTime: 58,
          numResults: 1,
          offset: 0,
          totalResults: 1,
          fuzzyLevel: 1
        },
        results: [
          {
            type: 'Point Address',
            id: 'US/PAD/p0/19173426',
            score: 14.51,
            address: {
              streetNumber: '15127',
              streetName: 'NE 24th St',
              municipalitySubdivision: 'Redmond',
              municipality: 'Redmond, Adelaide, Ames Lake, Avondale, Earlmount',
              countrySecondarySubdivision: 'King',
              countryTertiarySubdivision: 'Seattle East',
              countrySubdivisionCode: 'WA',
              postalCode: '98052',
              extendedPostalCode: '980525544',
              countryCode: 'US',
              country: 'United States Of America',
              countryCodeISO3: 'USA',
              freeformAddress: '15127 NE 24th St, Redmond, WA 980525544',
              countrySubdivisionName: 'Washington'
            },
            position: {
              lat: 47.6308,
              lon: -122.1385
            },
            viewport: {
              topLeftPoint: {
                lat: 47.6317,
                lon: -122.13983
              },
              btmRightPoint: {
                lat: 47.6299,
                lon: -122.13717
              }
            },
            entryPoints: [
              {
                type: 'main',
                position: {
                  lat: 47.6315,
                  lon: -122.13852
                }
              }
            ]
          }
        ]
      },
      () => geocoder.geocode('15127 NE 24th Street, Redmond, WA 98052')
    );

    const feature: GeocodingResult = result[0];
    expect(feature.name).toBe('15127 NE 24th St, Redmond, WA 980525544');
    expect(feature.center).toStrictEqual({ lat: 47.6308, lng: -122.1385 });
    expect(feature.bbox).toStrictEqual({
      _northEast: { lat: 47.6317, lng: -122.13717 },
      _southWest: { lat: 47.6299, lng: -122.13983 }
    });
    expect([[result]]).toMatchSnapshot();
  });
});
