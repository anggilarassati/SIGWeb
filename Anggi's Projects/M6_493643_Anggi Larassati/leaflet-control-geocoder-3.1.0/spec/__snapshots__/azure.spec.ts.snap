// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`L.Control.Geocoder.AzureMaps > geocodes 1`] = `
[
  [
    [
      {
        "bbox": {
          "_northEast": {
            "lat": 47.6317,
            "lng": -122.13717,
          },
          "_southWest": {
            "lat": 47.6299,
            "lng": -122.13983,
          },
        },
        "center": {
          "lat": 47.6308,
          "lng": -122.1385,
        },
        "name": "15127 NE 24th St, Redmond, WA 980525544",
      },
    ],
  ],
]
`;
